### ISSUE TITLE
Some backends are failing with parallel upload code

### ISSUE THREAD

**Issue by warwickmm** [duplicati/duplicati#3756]
I am encountering an issue where backups are failing for a few backends.  For small sources (less than 10 MB), all the following backends work without problems.  However, for a source around 300 MB, I see the following results:
* **Amazon Cloud Drive (doesn't work)**
* B2 (works)
* **Box (doesn't work)**
* Dropbox (works)
* Google Cloud Storage (works)
* Google Drive (works)
* Jottacloud (works)
* Local (works)
* Mega.nz (works)
* Microsoft OneDrive (works)
* Open Drive (works)
* SFTP (works)

In the failing cases, it appears that no data is ever uploaded and I eventually encounter an exception:
```
One or more errors occurred. (The channel is retired (The channel is retired) (One or more errors occurred. (Cannot access a disposed object. Object name: 'MobileAuthenticatedStream'.)))
```
Since the backups work for smaller sources, the key may be the case where all upload threads are begin used (4 threads at 50 MB volume size is less than 300 MB source).  However, setting `asynchronous-concurrent-upload-limit` to `1` does not fix the issue either.

This appears to have been introduced in revision 57b28182cb714c46226645b47fa80cb3d23f1f79 from pull request #3684.  I do not encounter this problem with revision 38dfdffb903a13a65169006b0d9444c25f92ef43 (first parent of  57b28182cb714c46226645b47fa80cb3d23f1f79).

**Comment by ts678** [comment]
Maybe a different issue (which I haven't tested on other backends or other variation), but I recently backed up a small file (to avoid a special case) then a Linux ISO, watched a whole lot of dblock "A task was cancelled", and wound up with a very incomplete set of dblock files on the backend, plus an interesting "look" in the database where the ISO wasn't kept as a file, yet the Block table still had its dblocks... Seeing any of that in your testing?

**Comment by warwickmm** [comment]
No, I haven't encountered that yet.  Is it easily reproducible?

**Comment by michaelmolino** [comment]
I am unable to backup to Amazon S3 with the following error:

> One or more errors occurred. (The channel is retired (The channel is retired) (One or more errors occurred. (Unable to write data to the transport connection: Connection reset by peer.)))

Google Drive works fine.  I've only just installed Duplicati so not sure about previous versions.

**Comment by warwickmm** [comment]
It looks like the `AmzCD` and `BoxBackend` classes are the only ones that use the [PostMultipartAsync](https://github.com/duplicati/duplicati/blob/83bb725dc6ed3010e0fefc0424d5d4e892af5178/Duplicati/Library/Backend/OAuthHelper/JSONWebHelper.cs#L132https://github.com/duplicati/duplicati/blob/83bb725dc6ed3010e0fefc0424d5d4e892af5178/Duplicati/Library/Backend/OAuthHelper/JSONWebHelper.cs#L132) method.  There may be an issue with the implementation there.

**Comment by ts678** [comment]
> this is happening to me using AWS S3, is this supposed to be fixed already??

What version are you on? Beta releases are still at a November 2018 bug level This looks fixed in Canary [v2.0.4.18-2.0.4.18_canary_2019-05-12](https://github.com/duplicati/duplicati/releases/tag/v2.0.4.18-2.0.4.18_canary_2019-05-12) via PR [Fix Amazon Cloud Drive and Box backends #3757](https://github.com/duplicati/duplicati/pull/3757), but why would it help S3? Issue author might have an opinion, or you might have a different issue than this one.

**Comment by warwickmm** [comment]
This specific issue was fixed.  Please open a new issue to describe the problem you're experiencing with S3.
